strawberry/experimental/pydantic/error_type.py

```diff
@@ -44,26 +49,57 @@ def field_type_to_type(type_):
 def error_type(
     model: Type[BaseModel],
     *,
-    fields: List[str],
+    fields: List[str] = None,
     name: Optional[str] = None,
     description: Optional[str] = None,
     federation: Optional[FederationTypeParams] = None,
+    all_fields: bool = False
 ):
     def wrap(cls):
         model_fields = model.__fields__
-        fields_set = set(fields)
+        fields_set = set(fields) if fields else set([])
 
-        cls = dataclasses.make_dataclass(
-            cls.__name__,
-            [
+        existing_fields = getattr(cls, "__annotations__", {})
+        fields_set = fields_set.union(
+            set(name for name, typ in existing_fields.items() if typ == auto)
+        )
+
+        if all_fields:
+            fields_set = set(model_fields.keys())
+
+        if not fields_set:
+            raise MissingFieldsListError(cls)
+
+        all_model_fields: List[Tuple[str, Any, dataclasses.Field]] = [
+            (
+                name,
+                get_type_for_field(field),
+                dataclasses.field(default=None),  # type: ignore
+            )
+            for name, field in model_fields.items()
+            if name in fields_set
+        ]
+
+        wrapped = _wrap_dataclass(cls)
+        extra_fields = cast(List[dataclasses.Field], _get_fields(wrapped))
+        private_fields = get_private_fields(wrapped)
+
+        all_model_fields.extend(
+            (
                 (
-                    name,
-                    get_type_for_field(field),
-                    dataclasses.field(default=None),  # type: ignore
+                    field.name,
+                    field.type,
+                    field,
                 )
-                for name, field in model_fields.items()
-                if name in fields_set
-            ],
+                for field in extra_fields + private_fields
+                if field.type != auto
```

Comment:
Why the check against auto here? Should we let private fields and strawberry fields use the auto type. It doesn't make much sense to use auto outside of the integrations IMO.

Reply:
The check against auto is to make sure we're not including the fields that we already captured above. If we don't do this we get duplication of the fields, as the get captured as extra fields and through the initial checks.

Reply:
Ah I see that makes sense, thanks.